pkg/agent/controller/networkpolicy/reject_test.go

```diff
@@ -502,3 +522,180 @@ func Test_getRejectPacketOutMutateFunc(t *testing.T) {
+			},
+		},
+		{
+			name: "IPv4,TCP,antreaProxy,serviceRemoteToExternal",
+			packetIn: genPacketIn(fakeSrcMac, fakeDstMac, fakeSrcIPv4, fakeDstIPv4, fakeSrcPort, fakeDstPort, fakeTCPSeqNum, false, true, []openflow15.MatchField{{
+				// EpSelectedRegMark
+				Class:          openflow15.OXM_CLASS_PACKET_REGS,
+				Field:          2,
+				HasMask:        false,
+				Length:         0,
+				ExperimenterID: 0,
+				Value: &openflow15.ByteArrayField{
+					Data:   []byte{0, 2, 0, 0, 0, 0, 0, 0},
+					Length: 64,
+				},
+				Mask: nil,
+			}}),
+			antreaProxyEnable: true,
+			expectFunc: func(mockClient *oftesting.MockClient, mockIStore *storetesting.MockInterfaceStore) {
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(nil, false)
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(nil, false)
+				mockClient.EXPECT().SendTCPPacketOut(fakeDstMac, openflow.GlobalVirtualMAC.String(), fakeDstIPv4, fakeSrcIPv4, gomock.Any(), gomock.Any(), false, fakeDstPort, fakeSrcPort, gomock.Any(), fakeTCPSeqNum+1, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(nil)
+			},
+		},
+		{
+			name:              "IPv4,UDP,antreaProxy,podLocal",
+			packetIn:          genPacketIn(fakeSrcMac, fakeDstMac, fakeSrcIPv4, fakeDstIPv4, fakeSrcPort, fakeDstPort, 0, false, false, []openflow15.MatchField{*openflow15.NewInPortField(uint32(fakeOFPort1))}),
+			antreaProxyEnable: true,
+			expectFunc: func(mockClient *oftesting.MockClient, mockIStore *storetesting.MockInterfaceStore) {
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeSIface, true)
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeDIface, true)
+				mockClient.EXPECT().SendICMPPacketOut(fakeDstMac, fakeSrcMac, fakeDstIPv4, fakeSrcIPv4, gomock.Any(), gomock.Any(), false, uint8(3), uint8(10), gomock.Any(), gomock.Any()).Times(1).Return(nil)
+			},
+		},
+		{
+			name:              "IPv4,TCP,antreaProxyDisable,podLocal",
+			packetIn:          genPacketIn(fakeSrcMac, fakeDstMac, fakeSrcIPv4, fakeDstIPv4, fakeSrcPort, fakeDstPort, fakeTCPSeqNum, false, true, nil),
+			antreaProxyEnable: false,
+			expectFunc: func(mockClient *oftesting.MockClient, mockIStore *storetesting.MockInterfaceStore) {
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeSIface, true)
+				mockIStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(fakeDIface, true)
+				mockIStore.EXPECT().GetInterfacesByType(gomock.Any()).Return([]*interfacestore.InterfaceConfig{fakeGWIface})
+				mockClient.EXPECT().SendTCPPacketOut(fakeDstMac, fakeSrcMac, fakeDstIPv4, fakeSrcIPv4, gomock.Any(), gomock.Any(), false, fakeDstPort, fakeSrcPort, gomock.Any(), fakeTCPSeqNum+1, gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Times(1).Return(nil)
+			},
+		},
```

Comment:
Move this before line 574, so that we can group test cases better.

Reply:
Done.